lib/index.js

```diff
@@ -314,12 +312,16 @@ export function buildJsx(tree, options) {
           const attribute = attributes[index]
 
           if (attribute.type === 'JSXSpreadAttribute') {
-            if (fields.length > 0) {
-              objects.push({type: 'ObjectExpression', properties: fields})
-              fields = []
+            const {argument} = attribute
```

Comment:
```suggestion
            const argument = attribute.argument
```

No need to destructure for one key

Reply:
Open to changing it back, I would note that personally I find de-structuring easier to read, and understand that a property is being accessed without being renamed.

Reply:
I lean towards using destructuring, but I don’t have strong feelings about this.

Reply:
- Most important for me, there’s a lot of precedence in this project and in all other unified projects to take props instead of destructure, if we’re going to change that It’s worth a separate discussion and feels like a refactor that can be separate?
- To have that discussion, generally, I don’t see a good reason to use newer syntax when old syntax works well, similarly for example, “normal” functions and such are super fine instead of arrow functions
- There are also benefits to taking props instead of destructuring:
  - you can do more, e.g., [`const value = child.value.replace(...)`](https://github.com/syntax-tree/estree-util-build-jsx/blob/6c35b60f0e1fb1b46f7b9bd4b22132ed469deffe/lib/index.js#L253-L265)
  - it’s cleaner to delve deeper, e.g., [`const attributes = node.openingElement.attributes`](https://github.com/syntax-tree/estree-util-build-jsx/blob/6c35b60f0e1fb1b46f7b9bd4b22132ed469deffe/lib/index.js#LL308C25-L308C25)
  - you’re not bound to a name some other project chose, or what makes sense on an object (a hypothetical `const nodeType = node.type`)
- There’s probably not a good reason to destructure *or* pick for one level: e.g., instead of `const {length} = list.length`, just use `list.length` everywhere. I believe TS can pick that up fine? E.g.
  ```js
  if (attribute.type === 'JSXSpreadAttribute') {
    fields.push(
      ...(attribute.argument.type === 'ObjectExpression'
        ? attribute.argument.properties
        : [{type: 'SpreadElement', argument: attribute.argument}])
    )
  }
  ```

Reply:
I’m not saying we should replace or use this everywhere. It’s just fine to use in some cases.

I see no need to make a distinction between older and newer syntax. Destructuring is part of the language and has been supported by all major platforms for a while now.

I think of not having to pick a name as a benefit (unless it doesn’t make sense).

I actually think it’s nice to pick for one level if the property is used several times.

Anyway, for me this is really a nit. I removed the picking of the argument altogether.

I didn’t use the ternary syntax that creates an intermediate array just for destructuring. IMO this looks too clever. An if-else statement looks simpler.